### PR TITLE
Query in url

### PR DESCRIPTION
**Pull request by joeinnes**
Use ?q=<search term>

Other params are left alone, and no action is taken if no parameter is present.

Works by putting the search term in the box and calling the search function.

### CODE REVIEW

**Comment by joeinnes** [comment]
Would close #13

File: primerpedia.js [hunk 1]
```diff
@@ -18,7 +18,7 @@
  *    and some templates are still displayed, so they need to be hidden via css
  */
 
-var apiUrl = "http://en.wikipedia.org/w/api.php?";
+var apiUrl = "https://en.wikipedia.org/w/api.php?";
```
Comment (waldyrious) [inline primerpedia.js]: Could you leave this out for now? Github automatically redirects waldyrious.github.io to my website waldyrious.net, which is not setup for https yet, so this API query would raise a mixed content error.

File: primerpedia.js [hunk 2]
```diff
@@ -62,4 +62,24 @@ function apiRequest(queryString){
 			}
 		}
 	});
-}
\ No newline at end of file
+}
+
+function getQueryVariable(variable) {
+	// http://stackoverflow.com/questions/2090551/parse-query-string-in-javascript
```
Comment (waldyrious) [inline primerpedia.js]: Thanks for adding documentation! I'd rather link directly to the [relevant answer](http://stackoverflow.com/a/2091331/266309), though.

File: primerpedia.js [hunk 3]
```diff
@@ -62,4 +62,24 @@ function apiRequest(queryString){
...
+    var query = window.location.search.substring(1);
```
Comment (waldyrious) [inline primerpedia.js]: Mixing spaces and tabs here :) please follow the current code style and use tabs for indentation.

File: primerpedia.js [hunk 4]
```diff
@@ -62,4 +62,24 @@ function apiRequest(queryString){
...
+        var pair = vars[i].split('=');
+        if (decodeURIComponent(pair[0]) == variable) {
+            return decodeURIComponent(pair[1]);
+        }
+    }
+    return null;
+}
+
+$(document).ready(function() {
+	if (getQueryVariable("q")) {
```
Comment (waldyrious) [inline primerpedia.js]: I'd rather be explicit and call the parameter "page". It's short enough, and self-explanatory.

File: primerpedia.js [hunk 5]
```diff
@@ -62,4 +62,24 @@ function apiRequest(queryString){
...
+		document.getElementById('search-term').value = getQueryVariable("q");
+		search();
+	}
+});
```
Comment (waldyrious) [inline primerpedia.js]: While you're at it, please add a newline at the end of the file :)

**Comment by waldyrious** [comment]
Thanks for the patch! I've added some inline comments, please take a look.

File: primerpedia.js (hunk 2 above)
Comment (joeinnes) [inline primerpedia.js]: Done.

File: primerpedia.js (hunk 3 above)
Comment (joeinnes) [inline primerpedia.js]: Thanks, updated with the latest commit.

File: primerpedia.js (hunk 4 above)
Comment (joeinnes) [inline primerpedia.js]: Ok, page it is then :)

File: primerpedia.js (hunk 5 above)
Comment (joeinnes) [inline primerpedia.js]: Done... although I'm not actually seeing it in the commit, strangely.

**Comment by joeinnes** [comment]
Thanks for all your feedback! Added a new commit which should iron stuff out.

**Comment by waldyrious** [comment]
You may have forgotten the indentation changes (or your editor played a trick on you :P)

**Comment by waldyrious** [comment]
But nevermind, that's a very minor change, so I'll merge and fix it myself. Thanks a bunch for the PR!

**Pull request merged**